### PR TITLE
[Spritelab] Implement new UI for user input block

### PR DESCRIPTION
**Pull request by ajpal**
Probably easier to review with [whitespace ignored](https://github.com/code-dot-org/code-dot-org/pull/38084/files?diff=unified&w=1)
No change to the user input functionality in sprite lab, just a new UI. The number indicates how many input prompts are in the queue. The input area overlays the spritelab playspace, aligned to the bottom and fitting the width of the playspace. The prompt text is truncated with ellipsis if needed to make sure the prompt fits on one line.
Before:
image

After:
Dec-02-2020 10-48-56

## Links

- [spec](https://docs.google.com/document/d/1pZppqxv7UoAQU8BOygErJRfbCHAzIeuDRB9yaFKY7-Q/edit#)
- [jira](https://codedotorg.atlassian.net/browse/STAR-1130)

## Testing story

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/p5lab/spritelab/SpritelabInput.jsx [hunk 1]
```diff
@@ -4,14 +4,53 @@ import React from 'react';
 import {popPrompt} from './spritelabInputModule';
 import * as coreLibrary from './coreLibrary';
 
+const styles = {
+  container: {
+    background: 'rgba(34, 42, 51, 0.85)',
```
Comment (maddiedierker) [inline apps/src/p5lab/spritelab/SpritelabInput.jsx:9]: is/should this be a color we reuse? if so, we should move this to our color constants file

File: apps/src/p5lab/spritelab/SpritelabInput.jsx [hunk 2]
```diff
@@ -25,26 +64,52 @@ class SpritelabInput extends React.Component {
...
-            value={this.state.userInput || ''}
-          />
-          <button type="button" onClick={() => this.userInputSubmit()}>
-            <i className="fa fa-check" />
-          </button>
+      <div id="spritelabInputArea" style={styles.container}>
+        <div style={styles.prompt}>
+          <span className="fa-stack">
+            <i className="fa fa-circle fa-stack-2x" />
+            <strong className="fa-stack-1x" style={{color: '#222A33'}}>
```
Comment (maddiedierker) [inline apps/src/p5lab/spritelab/SpritelabInput.jsx]: same comment re: using a color constant here

File: apps/src/p5lab/spritelab/SpritelabInput.jsx [hunk 3]
```diff
@@ -25,26 +64,52 @@ class SpritelabInput extends React.Component {
...
+            <input
+              style={styles.inputArea}
+              type="text"
+              onChange={event => this.setState({userInput: event.target.value})}
+              value={this.state.userInput || ''}
+            />
+            <button
+              style={styles.submitButton}
+              type="button"
+              onClick={() => this.userInputSubmit()}
```
Comment (maddiedierker) [inline apps/src/p5lab/spritelab/SpritelabInput.jsx]: super nit:
```suggestion
              onClick={this.userInputSubmit}
```

File: apps/src/p5lab/spritelab/SpritelabInput.jsx (hunk 1 above)
Comment (ajpal) [reply apps/src/p5lab/spritelab/SpritelabInput.jsx:9]: This color was specified by Mark in the spec, and it's darker than our darkest gray in the constants file. I think it's sort of one-off...

Comment (maddiedierker) [reply apps/src/p5lab/spritelab/SpritelabInput.jsx:9]: ah okay, if these colors are custom for this component then it's probably fine to leave their definitions in this file

**Review by maddiedierker — APPROVED** [review]

**Comment by ajpal** [comment]
Updated screenshot
image

If there are additional UI changes, I will handle those as follow-up PRs

**Pull request merged**